src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SamlSecurityToken.cs

```diff
@@ -0,0 +1,100 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.ObjectModel;
+using CoreWCF.Security;
+using MSIdentitySAML = Microsoft.IdentityModel.Tokens.Saml;
+
+namespace CoreWCF.IdentityModel.Tokens
+{
+    public class SamlSecurityToken : SecurityToken
+    {
+        private string _assertionXML;
+
+        protected SamlSecurityToken()
+        {
+        }
+
+        public SamlSecurityToken(MSIdentitySAML.SamlSecurityToken samlSecurityToken)
+        {
```

Comment:
Make this constructor and InternalSamlSecurityToken internal. Everything you need to know about a Saml token comes from the assertion, it wrapping the original IM.SamlSecurityToken is an internal implementation detail.